app/Http/Livewire/AllProjects.php

```diff
@@ -0,0 +1,236 @@
+<?php
+
+namespace App\Http\Livewire;
+
+use App\Enums\EngagementRecruitment;
+use App\Enums\MeetingType;
+use App\Enums\ProvinceOrTerritory;
+use App\Models\DisabilityType;
+use App\Models\Impact;
+use App\Models\Project;
+use App\Models\Sector;
+use Illuminate\Database\Eloquent\Builder;
+use Livewire\Component;
+use Livewire\WithPagination;
+use Spatie\LaravelOptions\Options;
+
+class AllProjects extends Component
+{
+    use WithPagination;
+
+    public string $query = '';
+
+    public array $statuses = [];
+
+    public array $seekings = [];
+
+    public array $seekingGroups = [];
+
+    public array $initiators = [];
+
+    public array $meetingTypes = [];
+
+    public array $locations = [];
+
+    public array $compensations = [];
+
+    public array $sectors = [];
+
+    public array $impacts = [];
+
+    public array $recruitmentMethods = [];
+
+    public function updatedStatuses()
+    {
+        if (! is_array($this->statuses)) {
+            return;
+        }
+        $this->statuses = array_filter($this->statuses,
+            function ($status) {
+                return $status != false;
+            }
+        );
```

Comment:
Could this be extracted into a method? Something like:

```php
public function filterItems(array $items) {
  return array_filter($items, function ($item) {
    return $item != false;
  });
}
```

And then you could call `$this->statuses = $this->filterItems($this->statuses);`

Reply:
I'm also curious how `false` would end up in an array? Let's discuss tomorrow, @chosww.

Reply:
@chosww Just wanted to check about this— I see you created the abstract method but I'm wondering if it's necessary? Can false values end up in these arrays?

Reply:
the $item parameter becomes false when I uncheck the checkbox, so I think what this array_filter is using callback value from the property update

Reply:
Oh, interesting. I'll take a look at that.